lib/rails/generators/channel/templates/channel.coffee.erb

```diff
@@ -0,0 +1,3 @@
+App.<%= channel_name %> = App.cable.subscriptions.create "<%= channel_class %>",
```

Comment:
I like that, Javan. Depends on having a cable JS initializer, but I think
we should have that anyway. +1 on putting this on Cable rather than App
too. Let's get that in and flow that through all the examples etc.

On Thu, Jul 9, 2015 at 2:15 PM, Javan Makhmali <notifications@github.com>
wrote:

> In lib/rails/generators/channel/templates/channel.coffee.erb
> <https://github.com/rails/actioncable/pull/17#discussion_r34248504>:
>
> > @@ -0,0 +1,3 @@
> > +App.<%= channel_name %> = App.cable.subscriptions.create "<%= channel_class %>",
>
> This will generate JavaScript that creates an error unless the user has
> already defined App.cable = Cable.createConsumer(...), and redeclaring a
> global @App in every file isn't ideal for those who don't wish to use
> that namespace.
>
> Perhaps we should standardize defining your consumer and subscriptions on
> @Cable itself. Something like:
>
> @Cable =
>   consumer: null
>   subscriptions: {}
>
> And the generator could generate something like:
>
> Cable.subscriptions.<%= channel_name %> = Cable.consumer?.subscriptions.create "<%= channel_class %>", ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/actioncable/pull/17/files#r34248504>.
>